composer/datasets/imagenet.py

```diff
@@ -86,3 +88,142 @@ def initialize_object(self, batch_size: int, dataloader_hparams: DataloaderHpara
+
+        if self.is_train:
+            split = 'train'
+            size = 100_000
+            n_shards = self.train_shards
+        else:
+            split = 'val'
+            size = 10_000
+            n_shards = self.val_shards
+        size = size - size % n_shards
+
+        if self.use_synthetic:
+            dataset = SyntheticBatchPairDataset(
+                total_dataset_size=size,
+                data_shape=[3, 64, 64],
+                num_classes=200,
+                num_unique_samples_to_create=self.synthetic_num_unique_samples,
+                device=self.synthetic_device,
+                memory_format=self.synthetic_memory_format,
+            )
+        else:
+            if self.is_train:
+                transform = transforms.Compose([
+                    transforms.RandomCrop(64, padding=8),
+                    transforms.RandomHorizontalFlip(),
+                    transforms.ToTensor(),
+                    transforms.Normalize(tinyimagenet_mean, tinyimagenet_std),
+                ])
+            else:
+                transform = transforms.Compose([
+                    transforms.ToTensor(),
+                    transforms.Normalize(tinyimagenet_mean, tinyimagenet_std),
+                ])
+
+            urls = ['/datasets/web_tinyimagenet200/%s_%05d.tar' % (split, i) for i in range(n_shards)]
+            size_per_device = size // dist.get_world_size()
+            dataset = WebDataset(urls, cache_dir=self.dataset_cache_dir,
+                                 cache_verbose=self.dataset_cache_verbose)
+            dataset = dataset.decode('pil').map_dict(jpg=transform).to_tuple('jpg', 'cls')
+            dataset = dataset.with_epoch(size_per_device).with_length(size_per_device)
+
+        return dataloader_hparams.initialize_object(dataset,
+                                                    batch_size=batch_size,
+                                                    sampler=None,
+                                                    drop_last=self.drop_last)
+
+
+@dataclass
+class WebImagenet1KDatasetHparams(WebDatasetHparams, SyntheticHparamsMixin):
+    """Defines an instance of the ImageNet-1K dataset for image classification.
+
+    Parameters:
+        resize_size (int, optional): The resize size to use. Defaults to -1 to not resize.
+        crop size (int): The crop size to use.
+        train_shards (int): Number of training split shards.
+        val_shards (int): Number of validation split shards.
+    """
+    resize_size: int = hp.optional("resize size. Set to -1 to not resize", default=-1)
+    crop_size: int = hp.optional("crop size", default=224)
+
+    train_shards: int = hp.optional('Training split shards', default=1024)
+    val_shards: int = hp.optional('Validation split shards', default=128)
+
+    def initialize_object(self, batch_size: int, dataloader_hparams: DataloaderHparams) -> DataLoader:
+        if self.is_train:
+            split = 'train'
+            size = 1_281_167
```

Comment:
Similar for all datasets, but I'm wondering if this should be set (or updated somewhere below) with the true size of the dataset (num_shards * samples_per_shard), otherwise there might be some inconsistency in how the Trainer measures time?